br: tikv startup failure when tikv v6.3.0 connect to pd v6.2.0

## Bug Report

### What version of TiKV are you using?
nightly build with older version pd v6.2.0

### What operating system and CPU are you using?
NA
### Steps to reproduce
1. deploy pd 6.2.v
2. deploy tikv 6.3 or nightly build
3. tikv start up failure
### What did you expect?

### What did happened?
tikv show the following error:
`2022/09/20 11:29:23.219 +00:00] [ERROR] [util.rs:505] ["request failed"] [err_code=KV:Pd:Grpc] [err="thread 'Grpc(RpcFailure(RpcStatus { code: 12-UNIMPLEMENTED, message: \"mainunknown method IsSnapshotRecovering for service pdpb.PD' panicked at '\", details: [] }))failed to get recovery mode from PD: Grpc(RpcFailure(RpcStatus { code: 12-UNIMPLEMENTED, message: "unknown method IsSnapshotRecovering for service pdpb.PD", details: [] }))"', ]`

--- a/components/server/src/server.rs
+++ b/components/server/src/server.rs
@@ -282,9 +282,16 @@ where
         let pd_client =
             Self::connect_to_pd_cluster(&mut config, env.clone(), Arc::clone(&security_mgr));
         // check if TiKV need to run in snapshot recovery mode
-        let is_recovering_marked = pd_client
-            .is_recovering_marked()
-            .expect("failed to get recovery mode from PD");
+        let is_recovering_marked = match pd_client.is_recovering_marked() {
+            Err(e) => {
+                warn!(
+                    "pd does not support snapshot recovery";
+                    "error" => ?e,
+                );
+                false
+            }
+            Ok(marked) => marked,
+        };
 
         if is_recovering_marked {
             // Run a TiKV server in recovery modeß

--- a/components/server/src/server.rs
+++ b/components/server/src/server.rs
@@ -285,7 +285,7 @@ where
         let is_recovering_marked = match pd_client.is_recovering_marked() {
             Err(e) => {
                 warn!(
-                    "pd does not support snapshot recovery";
+                    "failed to get recovery mode from PD";
                     "error" => ?e,
                 );
                 false

